src/librustc_typeck/check/pat.rs

```diff
@@ -1391,15 +1400,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
     /// Type check the length of an array pattern.
     ///
-    /// Return the length of the variable length pattern,
+    /// Returns the length of the variable length pattern,
     /// if it exists and there are no errors.
+    ///
+    /// In case the pattern has a fixed length and the array
+    /// length is not yet determined, returns the array type
+    /// with the inferred length.
     fn check_array_pat_len(
         &self,
         span: Span,
+        inner_ty: Ty<'tcx>,
```

Comment:
We usually refer to this as the "element type".

Reply:
So in that case:
```suggestion
        elem_ty: Ty<'tcx>,
```